single_number/single_number.py

```diff
@@ -2,10 +2,19 @@
 Input: a List of integers where every int except one shows up twice
 Returns: an integer
 '''
+from collections import Counter
 def single_number(arr):
     # Your code here
-
-    pass
+    index = 0
+    arr.sort()
+    while index < len(arr) - 1:
+        if arr[index] == arr[index+1] and index < len(arr)-3:
+            del arr[index:index+2]
+        elif arr[index] == arr[index+1]:
```

Comment:
This solution breaks for the case [1,1,2,2,3,3,4] 
Better way is using xor operator ,
```python
def singleNumber2(self, nums):
    res = 0
    for num in nums:
        res ^= num
    return res
```